include/mbedtls/x509.h

```diff
@@ -500,6 +500,23 @@ int mbedtls_x509_info_cert_type(char **buf, size_t *size,
 int mbedtls_x509_info_key_usage(char **buf, size_t *size,
                                 unsigned int key_usage);
 
+/**
+ * \brief          This function parses a CN string as an IP address.
+ *
+ * \param cn       The CN string to parse. CN string MUST be NUL-terminated.
```

Comment:
```suggestion
 * \param cn       The CN string to parse. CN string MUST be NULL-terminated.
```

Reply:
NULL is a macro for a null pointer, whereas NUL is the name given to the first ASCII character that we're expecting here.

Reply:
The word *[null](https://dictionary.cambridge.org/dictionary/english/null)* is an English word, meaning (in this context) “having the value 0”. Since it's an ordinary English word, it's typeset in the normal font and all lowercase (or with the initial letter capitalized if it's at the beginning of a sentence). Strings in C are “null-terminated” since the last character is 0.

The symbol [`NULL`](https://en.cppreference.com/w/c/types/NULL) is a shortcut for “a null pointer” which is commonly used in C documentation. “[Null pointer](https://en.cppreference.com/w/c/language/pointer#Null_pointers)” is a technical term which is defined by the C language specification. (Strictly speaking there can be null pointers that are distinguishable from `NULL`, but if `p` is a null pointer, then `p == NULL` has to evaluate to 1: all null pointers are equal). This is a programming symbol, so it's typeset in a code font and is case-sensitive.

[NUL](https://en.wikipedia.org/wiki/ASCII#ASCII_control_code_chart) is the abbreviation of “null” in ASCII. (An older, forgotten version of ASCII spelled it NULL.) This is an abbreviation, but not a symbol, so it's typeset in the normal text font, but all caps.

The single-word phrase “null” is the name of [Unicode character](https://www.unicode.org/charts/PDF/U0000.pdf) U+0000. This is an English phrase, so it's typeset in the normal text font. Unicode character names are usually typeset in all caps, but they don't have to be.

Reply:
This still needs to be updated following the discussion. BUT: we are coming to the conclusion elsewhere that these sort of functions should actually take (pointer, length) rather than requiring null-terminated strings!

```suggestion
 * \param cn       The CN string to parse. CN string MUST be null-terminated.
```

Reply:
> This still needs to be updated following the discussion. BUT: we are coming to the conclusion elsewhere that these sort of functions should actually take (pointer, length) rather than requiring null-terminated strings!

I'm not adding a new function here, just moving an existing one, so I'll leave the API as-is for this PR.